lib/src/game_storage.dart

```diff
@@ -50,27 +50,28 @@ class GameStorage {
     targetPlatform.clearValues();
   }
 
-  int _getIntValue(String key, [int defaultValue = 0]) {
-    assert(key != null);
-
+  int? _getIntValue(String key, [int? defaultValue = 0]) {
     final strValue = targetPlatform.getValue(key);
     return _parseValue(strValue, defaultValue);
   }
 
   void _setIntValue(String key, int value) {
     assert(key != null);
-    final val = (value == null) ? null : value.toString();
-    targetPlatform.setValue(key, val);
+    if (value == null) {
+      throw UnimplementedError();
```

Comment:
I'd expect analyzer to  yell about this! Will never be hit!

Reply:
The spec does not require it, only "It is a warning to use a null aware operator (?., ?.., ??, ??=, or ...?) on an expression of type T if T is strictly non-nullable."

Reply:
I'd expect this to be a dead code block – like `if(false) {}` – type hint from the analyzer.

Not arguing about what the spec says or doesn't – just what I'd expect as a user. 😄

Reply:
Yeah, we don't specify anything about dead code, but seems like a good analyzer hint/lint.

Reply:
With https://dart-review.googlesource.com/c/sdk/+/137641 analyzer will report such hints.
![image](https://user-images.githubusercontent.com/384794/75482793-5a971180-595a-11ea-9c5d-b1675c41a42f.png)
```
info: The operand can't be null, so the condition is always true. (dead_null_condition_true at [pop_pop_win] lib/src/game_storage.dart:59)
info: The operand can't be null, so the condition is always false. (dead_null_condition_false at [pop_pop_win] lib/src/game_storage.dart:60)
```

Reply:
Nice! https://github.com/dart-lang/sample-pop_pop_win/pull/67/commits/a1c7ad4ac88067bb41e8ba58b9f07f0c14a9c5e0